lib/kafo/hooking.rb

```diff
@@ -48,11 +48,17 @@ def loaded?
     def execute(group, log_stage: true)
       logger = Logger.new(group)
       logger.notice "Executing hooks in group #{group}" if log_stage
-      self.hooks[group].keys.sort_by(&:to_s).each do |name|
+
+      sorted_hooks = self.hooks[group].keys.sort do |a, b|
+        File.basename(a.to_s) <=> File.basename(b.to_s)
```

Comment:
After reading about the enumerable sort() function and the `<=>` operator in Ruby, I think I finally understand how this is working. 

I used the following to test:

```
irb(main):042:0> myhash = {'fookey' => 'foo', 'barkey' => 'bar', 'bazkey' => 'baz'}
=> {"fookey"=>"foo", "barkey"=>"bar", "bazkey"=>"baz"}
irb(main):043:0> myhash.keys.sort do |a, b|
irb(main):044:1*   puts a
irb(main):045:1>   puts b
irb(main):046:1>   a <=> b
irb(main):047:1> end
fookey
barkey
barkey
bazkey
fookey
bazkey
=> ["barkey", "bazkey", "fookey"]
irb(main):048:0> myhash.keys.sort do |a, b|
irb(main):049:1*   puts a
irb(main):050:1>   puts b
irb(main):051:1>   1
irb(main):052:1> end
fookey
barkey
barkey
bazkey
=> ["bazkey", "barkey", "fookey"]
irb(main):053:0> myhash.keys.sort do |a, b|
irb(main):054:1*   puts a
irb(main):055:1>   puts b
irb(main):056:1>   0
irb(main):057:1> end
fookey
barkey
barkey
bazkey
=> ["fookey", "barkey", "bazkey"]
irb(main):058:0> myhash.keys.sort do |a, b|
irb(main):059:1*   puts a
irb(main):060:1>   puts b
irb(main):061:1>   -1
irb(main):062:1> end
fookey
barkey
barkey
bazkey
=> ["fookey", "barkey", "bazkey"]
```